nixpkgs/nixpkgs.bzl

```diff
@@ -1205,19 +1205,18 @@ def _cp(repository_ctx, src, dest = None):
         ])
 
     # Copy the file
-    # TODO: auto detect the executable bit of the source file
-    # It is set to False (i.e. not executable) which was the previous behavior.
-    # It means that executable files will miss the bit and won't be executable.
-    # Forcing it to True will fix this behavior, but will impact a lot of file
-    # (most of the time, files do not have the executable bit) and this may
-    # lead to other errors in the next process are checking for executable bit.
-    # One other side effect of this change is that you will have a difference
-    # in the nix hash computed when nix is run by rules_nixpkgs or directly.
     repository_ctx.file(repository_ctx.path(dest),
                         repository_ctx.read(repository_ctx.path(src)),
                         executable=False,
                         legacy_utf8=False)
 
+    # Copy the executable bit of the source
+    # This is important to ensure that copied binaries are executable.
+    repository_ctx.execute([repository_ctx.which("chmod"),
+                            "--reference",
+                            repository_ctx.path(src),
+                            repository_ctx.path(dest)])
+
```

Comment:
This change adds a new requirement to `nixpkgs_local_repository`. Before, it only used the `repository_ctx` API and didn't make any further assumptions about the environment. Now, it assumes that `chmod` is in `PATH`. This may not be the case on Windows. A simple way to avoid this issue would be to add a check like the following and simply skip the permission setting on Windows.
```
if get_cpu_value(repository_ctx) == "x64_windows:
```

Reply:
I've added the proposed check.